### PR TITLE
Documentation Improvements

### PR DESCRIPTION
**Pull request by ac360**
## Description:

**Status:** WIP

This PR introduces some minor documentation improvements.  Here are its goals:

- [x] Add in any missing pieces from the previous documentation.
- [x] Create a `serverless.yml` reference document which lists all available properties of `serverless.yml` in one place.
- [x] Clean up grammar, structural, typo mistakes.
- [x] Reign in new Guide documents (i.e., version pinning -> services) and put them in sections that already exist for them.
- [ ] Make sure each CLI command has a separate document.  Instead of burying commands in the documents of others.

## Todos:

- [ ] Write documentation
- [ ] Change ready for review message below

***Is this ready for review?:*** NO

### CODE REVIEW

File: docs/providers/aws/guide/serverless.yml.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+provider:
+  name: aws
+  runtime: nodejs4.3
+  stage: dev # Set the default stage used. Default is dev
+  region: us-east-1 # Overwrite the default region used. Default is us-east-1
+  profile: production # The default profile to use with this service
+  memorySize: 512 # Overwrite the default memory size. Default is 1024
+  timeout: 10 # The default is 6
+  deploymentBucket: com.serverless.${self:provider.region}.deploys # Overwrite the default deployment bucket
```
Comment (mthenw) [inline docs/providers/aws/guide/serverless.yml.md]: IMHO it's worth to rephrase to "Deployment bucket name. Default is generated by the framework".

File: docs/providers/aws/guide/testing.md [hunk 2]
```diff
@@ -12,17 +12,21 @@ layout: Doc
...
-- unit testing should be used as a main tool for verifying business logic,
-- integration tests should be used for verifying integration with other services.
+* Write your business logic so that it is separate from your FaaS provider (e.g., AWS Lambda), to keep it provider-independent, reusable and more easily testable.
+* When your business logic is written separately from the FaaS provider, you can write traditional Unit Tests to ensure it is working properly.
+* Write Integration Tests to verify integrations with other services are working correctly.
 
-Let's take simple Node.js function as an example. The responsibility of this function is to save user into DB and send welcome email:
+## A Poor Example
+
+Here is an example in Node.js of how to follow the practices above.  This job this Function should perform is to save a user in a database and then send a welcome email:
```
Comment (mthenw) [inline docs/providers/aws/guide/testing.md]: "This job this Function"

File: docs/providers/aws/guide/testing.md [hunk 3]
```diff
@@ -77,22 +81,30 @@ class Users {
...
+Now, the above class keeps business logic separate.  Further, the code responsible for setting up dependencies, injecting them, calling business logic functions and interacting with AWS Lambda is in its own file, which will be changed less often.  This way, the business logic is not provider dependent, easier to re-use, and easier to test.
+
+Further, this code doesn't require running any external services.  Instead of a real `db` and `mailer` services, we can pass mocks and assert if `saveUser` and `sendWelcomeEmail` has been called with proper arguments.
+
+Unit Tests can easily be written to cover the above class.  Then invoke the function (`serverless invoke`) with fixture email address, check if user is actually saved to DB and check if email was received to see if everything is working together.
```
Comment (mthenw) [inline docs/providers/aws/guide/testing.md]: IMHO it's worth to add that `serverless invoke` is helpful "integration tests". I mean we should use "integration tests" phrase :)

**Comment by mthenw** [comment]
@ac360 what about links. There is many links that point to 404? At least when browsed in Github.

**Comment by pmuens** [comment]
Thanks for reviewing @mthenw! I updated the docs according to your feedback!

Do you have an example for 404 links?

**Comment by mthenw** [comment]
@pmuens 

## 404

- https://github.com/serverless/serverless/tree/master/docs/providers/aws - "AWS system credentials" 
- https://github.com/serverless/serverless/blob/master/docs/providers/aws/events/s3.md - "Serverless Variable syntax"
- https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/deploying.md - "deploy command docs"
- 

## Links that points to website but should point to .md

in .md file we should link to .md files. There is build step already that transform them.

- https://github.com/serverless/serverless/tree/master/docs/providers/aws/cli-reference - "AWS system credentials"
- https://github.com/serverless/serverless/tree/master/docs/providers/aws/events - "AWS system credentials"
- https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/credentials.md - "variables"
-

**Pull request merged**